### PR TITLE
 adding labels to options of codelist via the codelist edit/create form

### PR DESCRIPTION
**Pull request by lagartoverde**
Implements having labels for codelist, which is a summary for the value.
- issue:  [GN-4257](https://binnenland.atlassian.net/browse/GN-4257)
- related Figma: see [GN-4275](https://binnenland.atlassian.net/browse/GN-4275) and [figma](https://www.figma.com/file/hqKonrXwkHrLNG0WfWqKmj/RB-variables-and-features?type=design&node-id=119-1257)
  - **!! Figma is outdated and lots of extra changes were added !!** use figma for general design, but not for user experience.
- connected backend PR (**run this**): [app-RB #52](https://github.com/lblod/app-reglementaire-bijlage/pull/52).
- this change also means a change in the variable plugin: [editor-plugins #146](https://github.com/lblod/ember-rdfa-editor-lblod-plugins/pull/146) (variable plugin can be updated later and is not coupled with this PR)

### Implements
- full restructure of pages for codelists. Includes a lot more user feedback that clearly shows what happens/what is wrong. => as they are pretty much new pages, looking at the file itself (instead of changes) is probably easier.
- adds the possibility to also add labels
  - labels are not mandatory (never)
  - labels should be unique per codelist
- fixes breadcrumbs => influence on whole app, not just codelist pages
- to avoid pagenation issues, a max of 20 codelist options are accepted
  
### Implementation Details
- `did-insert` is still used, but I think this is (kinda) deprecated. I am not sure if we want to keep using this or not. For now I left it in (because it is easy to use :shrug: )
#### ChangesetList
ember-changeset does not handle hasMany relationships. this resulted in a lot of boilerplate code on component level (that also introduced quite a few bugs, because it is complex boilerplate code). I tried to put this all in a new class ChangesetList, which should also more cleanly handle everything, decreasing bugs.
- the implementation is not perfect and could definitely use some more work for more general usage. This is just a first iteration that already improves over what was done before and is already useful just for this one usecase.

### CODE REVIEW

**Comment by x-m-el** [comment]
-placeholder-

**Comment by x-m-el** [comment]
second reviewer: @lagartoverde (can't add officially because github :shrug: )

**Comment by x-m-el** [comment]
Todo:
Add a "are you sure you want to leave" pop-up when a user leaves the edit page with unsaved changes. 
Pop-up doesn't have to come if there are no changes to save.

**Comment by x-m-el** [comment]
This PR also fixes https://binnenland.atlassian.net/browse/GN-4308 by limiting amount of paginations.
- wait for the comment in https://binnenland.atlassian.net/browse/GN-4308 to be resolved ("is this a good fix for this issue" question). Because if the answer is "no", code here should be changed.
- Note that the limit is set for *new* codelists. Old codelists still have this problem. **How do we fix old codelists that might have more than 20 options? Should we just check if these exist and handle them case by case (instead of a general migration)?**

File: app/components/codelist-form.hbs [hunk 1]
```diff
@@ -1,166 +1,198 @@
...
+      </AuLabel>
+      <PowerSelect
+        @allowClear={{false}}
+        @searchEnabled={{false}}
+        @options={{this.codelistTypes}}
+        @selected={{this.selectedType}}
+        @onChange={{this.updateCodelistType}} as |type|
+      >
+        <span class='codelist-type-select-option'>
+          {{t (concat 'codelist.attr.types.' type.label)}}
```
Comment (elpoelma) [inline app/components/codelist-form.hbs:68]:
```suggestion
          {{t (concat 'codelist.attr.types.' type.value)}}
```

**Review by elpoelma — CHANGES_REQUESTED** [review]
Seems to work well overall. Just noticed some translation issues mentioned in my comments. 
The ChangesetList seems like a good enough solution to tracking `hasMany` changes, maybe it could even be updated to support `hasMany-hasMany` relationships.

**Comment by x-m-el** [comment]
> Todo: Add a "are you sure you want to leave" pop-up when a user leaves the edit page with unsaved changes. Pop-up doesn't have to come if there are no changes to save.

I can't seem to do this cleanly for the general usecase. Only letting it show up when a user clicks cancel without saving is a possibility, but I fear this users will then lose unplanned progress by *thinking* it will also ask when they leave the page by other means.
So I think by default not doing anything is better then, as at least it is consistent...

**Comment by x-m-el** [comment]
added some store cleanup (no "new" code) and checked the prod database: as far as I can tell no codelists exist yet :shrug: 
Merging :)

**Pull request merged**